src/network/netdev/macsec.c

```diff
@@ -101,8 +101,13 @@ static void macsec_receive_channel_free(ReceiveChannel *c) {
                 return;
 
         if (c->macsec) {
-                if (c->sci.as_uint64 > 0)
-                        ordered_hashmap_remove(c->macsec->receive_channels, &c->sci.as_uint64);
+                if (c->sci.as_uint64 > 0) {
+                        ReceiveChannel *stored;
+
+                        stored = ordered_hashmap_get(c->macsec->receive_channels, &c->sci.as_uint64);
+                        if (stored == c)
+                                ordered_hashmap_remove(c->macsec->receive_channels, &c->sci.as_uint64);
```

Comment:
hmm, for the regular hashmaps there's `hashmap_remove_value()` which makes this comparison implicitly. I think we should just add a similar helper call `ordered_hashmap_remove_value()` and use that here.

Reply:
Done. PTAL.